Explain Nextflow config order of priority (#279)
seqeralabs/docs: platform_versioned_docs/version-24.1/launch/advanced.mdx

```ini
    manifest {
        name = 'My_RNASeq_Pipeline'
        description = 'Generates RNASeq results using a test profile'
    }
```

#### Nextflow configuration order of priority

When launcing pipelines in Platform, Nextflow configuration values can be supplied from the `nextflow.config` file in the pipeline repository and the **Nextflow config file** field in the pipeline launch form. If different values of the same configuration parameter are defined, Nextflow parameters defined in the launch form **Nextflow config file** field override the same parameters in your `nextflow.config` file. 

Configuration values set in the **Global Nextflow config** field during compute environment creation are pre-filled in the **Nextflow config file** field during pipeline launch. These pre-filled values from the compute environment can be overridden manually during launch. 

| Priority | Nextflow configuration                               |
|----------|------------------------------------------------------|
| Highest  | Pipeline launch form **Nextflow config file** field  |

> | Medium   | Compute environment **Global Nextflow config** field |

Author: ```suggestion
|                | Compute environment **Global Nextflow config** field |
```

    }
```

#### Nextflow configuration order of priority

> When launcing pipelines in Platform, Nextflow configuration values can be supplied from the `nextflow.config` file in the pipeline repository and the **Nextflow config file** field in the pipeline launch form. If different values of the same configuration parameter are defined, Nextflow parameters defined in the launch form **Nextflow config file** field override the same parameters in your `nextflow.config` file. 

Reviewer: So it looks like by default, the repo values are used, unless
- the user specifies an override globally for the compute env
- the user specifies an override in the launch form, possibly overriding what is set for the compute env

I wonder if a table might help with clarity here? Overrides are always kind of complex to explain. Reminds me of overrides for Opscode Chef automation application.

Author: @jason-seqera does the example below illustrate the order clearly? While a table may work better, columns or rows for each scenario could be long and unwieldy.

Reviewer: @llewellyn-sl, I had to-read it to understand, it looks like I do (?) how it works. But it might be easier to parse if it's presented in an alternate format. Maybe NF users are familiar with this enough that it is clear as is.

Configuration values set in the **Global Nextflow config** field during compute environment creation are pre-filled in the **Nextflow config file** field during pipeline launch. These pre-filled values from the compute environment can be overridden manually during launch. 

For example, if:

1. The `nextflow.config` file in your pipeline repository contains this manifest: 

> ```ini

Reviewer: This isn't showing up as indented below the 1). Or for 2) or 3) below.

>     ```ini

Reviewer: We could also specify a title here: https://docs.constellationnetwork.io/markdown/code-blocks/#code-title

Maybe _Repository nextflow.config_, _Compute environment global config_, _pipeline launch form config_ or similar respectively.

Author: Good idea! Done.

>     manifest {

Reviewer: This has an extra 4 spaces of indentation; Does it make sense for this to be flush left?

        name = 'A'

>         description = 'Description A'

Reviewer: I wonder if something like

> pipeline repository example description

or something more concrete might be helpful for this?

    }
```

#### Nextflow configuration order of priority

When launcing pipelines in Platform, Nextflow configuration values can be supplied from the `nextflow.config` file in the pipeline repository and the **Nextflow config file** field in the pipeline launch form. If different values of the same configuration parameter are defined, Nextflow parameters defined in the launch form **Nextflow config file** field override the same parameters in your `nextflow.config` file. 

Configuration values set in the **Global Nextflow config** field during compute environment creation are pre-filled in the **Nextflow config file** field during pipeline launch. These pre-filled values from the compute environment can be overridden manually during launch. 

> | Nextflow configuration order of priority (highest to lowest) |
> | ------------------------------------------------------------ |
> | Pipeline launch form **Nextflow config file** field |
> | Compute environment **Global Nextflow config** field |
> | Pipeline repository `nextflow.config` file |

Reviewer 2: Could we have a second column here for priority?
```suggestion
| Priority | Nextflow configuration                               |
|----------|------------------------------------------------------|
| Highest  | Pipeline launch form **Nextflow config file** field  |
| Medium   | Compute environment **Global Nextflow config** field |
| Lowest   | Pipeline repository `nextflow.config` file           |
```

For example, if:

1. The `nextflow.config` file in your pipeline repository contains this manifest: 

    ```ini title="Pipeline repository config file"
    manifest {
        name = 'A'
        description = 'Pipeline description A'
        mainScript = 'main.nf'

>         defaultBranch = 'master'

Reviewer 2: I think you can remove these two throughout this PR. We include them in nf-core for paranoia and safety, but they're generally not needed. We may even deprecate `defaultBranch` soon-ish I think.

Run custom code either before or after the execution of the Nextflow script. These fields allow you to enter shell commands.

Pre-run scripts are executed in the nf-launch script prior to invoking Nextflow processes. Pre-run scripts are useful for:
- Specifying an alternate Nextflow version to use for the run:

  ```bash 

>   nextflow self-update

Reviewer 3: this is only need when updating a version prior to `24.10.0` to version `24.10.0` (or later)

Reviewer 2: @pditommaso you think it's ok to keep in for now, then in six months once everyone has the newer nf-launcher with Nextflow >= 24.10.0 we can remove again?

@llewellyn-sl maybe we can add a comment in the docs to this effect.

I don't think that having this here when it is not needed will have any negative effect.

Reviewer 3: I agree, it's fine

Run custom code either before or after the execution of the Nextflow script. These fields allow you to enter shell commands.

Pre-run scripts are executed in the nf-launch script prior to invoking Nextflow processes. Pre-run scripts are useful for:
- Specifying an alternate Nextflow version to use for the run:

    ```bash 
    nextflow self-update
    export NXF_VER=24.10.0
    ```
    :::info

>     `nextflow self-update` is required before updating a pre-24.10.0 version of Nextflow to version 24.10.0 or later. 

Reviewer 2: ```suggestion
    `nextflow self-update` is only required when updating a pre-24.10.0 version of Nextflow to version 24.10.0 or later. 
```
